src/componets/Clock.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+type Props = {
+  name: string
+};
+
+type State = {
+  time: Date
+};
+
+export class Clock extends React.PureComponent<Props, State> {
+  state = {
+    time: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState((currentTime) => ({
+        ...currentTime,
+        time: new Date(),
+      }));
+
+      // eslint-disable-next-line no-console
+      console.info(this.state.time.toUTCString().slice(-12, -4));
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Props): void {
+    if (prevProps.name !== this.props.name) {
+      // eslint-disable-next-line no-console
+      console.debug(`Renamed from ${prevProps.name} to ${this.props.name}`);
+    }
+  }
+
+  componentWillUnmount(): void {
+    window.clearInterval(this.timerId);
+  }
+
+  render() {
+    const { name } = this.props;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {name}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {this.state.time.toUTCString().slice(-12, -4)}
```

Comment:
please don't call a function from the interface because it will be executed every time when the state change, so make sense to move it to the variable